src/main/java/teammates/logic/backdoor/BackDoorLogic.java

```diff
@@ -153,8 +154,17 @@ public String persistDataBundle(DataBundle dataBundle)
         }
         frDb.createFeedbackResponses(responses.values());
 
+        Set<String> sessionStrings = new HashSet<>();
+        
         for (FeedbackResponseAttributes response : responses.values()) {
-            updateRespondants(response.feedbackSessionName, response.courseId);
+            
+            String sessionString = response.feedbackSessionName + " % " + response.courseId;
+            
+            if (!sessionStrings.contains(sessionString)) {
+                updateRespondants(response.feedbackSessionName, response.courseId);
+            }
+            
+            sessionStrings.add(sessionString);
```

Comment:
Consider this:
```java
if (!sessionStrings.contains(sessionString)) {
    updateRespondants(response.feedbackSessionName, response.courseId);
    sessionStrings.add(sessionString);
 }
```
or this 
```java      
if (sessionStrings.contains(sessionString)) {
    continue;
}

updateRespondants(response.feedbackSessionName, response.courseId);
sessionStrings.add(sessionString);
```

The choice can be based on what you consider to be the 'normal flow'. Try to keep the normal flow un-nested and deal with 'abnormal' conditions using branches.